docs/source/geopmexporter.1.rst

```diff
@@ -0,0 +1,194 @@
+is the domain index.  Provide the "``*``" character as the third string to
+request all domains available on the system.
+
+
+GEOPM Summary Method
+--------------------
+
+There are two options available for summarizing the statistics for the
+Prometheus exporter.  The default option is to use the GEOPM ``StatsCollector``
+interface to summarize sampled values.  This is a high performance option that
+provides fine grained summary statistics through a Prometheus Gauge metric
+interface:
+
+count:
+   Number of samples recorded since last Prometheus server scrape
+
+first:
+   First value sampled after last Prometheus server scrape
+
+last:
+   Most recently sampled value prior to Prometheus server scrape
+
+min:
+   Smallest sampled value since last Prometheus server scrape
+
+max:
+   Largest sampled value since last Prometheus server scrape
+
+mean:
+   Sum of all samples divided by the number of samples since last Prometheus
+   server scrape
+
+std:
+   Sample standard deviation represented by the following sum over samples:
+
+       ``sqrt(sum((x_i - m) ^ 2) / (n - 1))``
+
+   where ``m`` is the mean, ``x_i`` are the samples, and ``n`` is the number of
+   samples
+
+
+Prometheus Summary Method
+-------------------------
+
+When the user specifies the ``--summary prometheus`` command line option the
+``Summary`` and ``Counter`` metrics built into Prometheus are used to aggregate
+samples over time.  In this way the batch of signals read from the GEOPM is used
+to update a ``Summary`` metric for GEOPM signals with variable behavior, or a
+``Counter`` metric for signals with monotone behavior.  Each metric provides
+slightly different statistics for the Prometheus server on the port opened by
+``geopmexporter``.  the ``Summary`` metric exports the **count** and **sum**
+statistics, where as the ``Counter`` metric exports the **count** and **total**
+statistics.  GEOPM calls into the Prometheus metrics through a Python interface.
+This means that the quartile metrics are not currently supported for ``Summary``
+metrics. See the Prometheus documentation for more information about the
+``Summary`` and ``Counter`` metrics and the statistics that they export to the
+Prometheus server.
+
+Using Prometheus to aggregate samples provides an easy method to compare against
+the GEOPM ``StatsCollector`` method for summary.  Using the ``--summary prometheus``
+mode provides a more course grained view of the metrics gathered, and also tends
+to have a higher system overhead.
+
+
+Systemd Service
+---------------
+
+One simple way to manage the GEOPM Prometheus exporter process is to create a
+systemd service.  To use ``geopmexporter`` with systemd create a systemd unit
+configuration file called ``geopmexporter.service`` with the following content:
+
+.. code-block:: ini
+
+   [Unit]
+   Description=Prometheus GEOPM Exporter
+   StartLimitIntervalSec=0
+   StartLimitBurst=3
+   Wants=geopm.service
```

Comment:
Will the exporter still function if geopm.service is not available?  Wondering if this should be ``Requires`` instead.

Reply:
In theory, you could be exporting only signals that can be read without the service.  Not very likely, but could be a supported use case (thinking sysfs entries for gaudi that are not privileged.